assignments/array-methods.js

```diff
@@ -51,33 +55,77 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
-console.log(allCaps); 
+// The event director needs to have all the runner's first names converted to uppercase because 
+//the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
+
+const allCaps = runners.map(currentValue => currentValue.first_name.toUpperCase());
+
+
+// console.log(allCaps); 
+
+
+
 
 // ==== Challenge 3: Use .filter() ====
-// The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
-console.log(largeShirts);
+// The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with 
+//large sized shirts so they can choose a different size. Return an array named largeShirts that contains 
+//information about the runners that have a shirt size of L and log the result
+
+let largeShirts = runners.filter(currentValue => currentValue.shirt_size === "L");
+
+
+// console.log(largeShirts);
+
+
+
+
 
 // ==== Challenge 4: Use .reduce() ====
-// The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
-console.log(ticketPriceTotal);
+// The donations need to be tallied up and reported for tax purposes. Add up all the donations into a 
+//ticketPriceTotal array and log the result
+
+let ticketPriceTotal = runners.reduce((total, currentValue) => {
+    return total += currentValue.donation;
+}, 0);
+
+// console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
-// Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
+// Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of 
+//potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then 
+//solve 3 unique problems using one or many of the array methods listed above.
 
 // Problem 1
+//As a token of our appreciation, we want to make sure our top doners receive a special 'Thank-You' card. Sort out all of the doners who donated $150 or more.
+const topDoners = runners.filter(currentValue => currentValue.donation > 150);
+
+// console.log(topDoners);
+
+
 
 // Problem 2
+//Using .forEach and template literals, create a list of the participants first and last name, as well as their email.
+let contactInfo = runners.forEach(function(currentValue){
+    // console.log(`${currentValue.first_name} ${currentValue.last_name}, ${currentValue.email}`);
+});
+
+
+// Problem 3
+let fidel = runners.filter(currentValue => currentValue.last_name === "Fidel");
```

Comment:
```suggestion
let fidel = runners.filter(currentValue => currentValue.first_name === "Fidel");
```
Fidel is someone's first name in the array